### DISCUSSION TITLE
Register a controlled component using Controller or useController

### DISCUSSION THREAD

**Discussion by apedretti** [react-hook-form/react-hook-form#4916] [Q&A] [answered]
I'm crafting a web application that has a few custom components. Since I'm using RHF for the forms, I'm trying to integrate those custom, controlled components with RHF.

As I understood from the [online example](https://react-hook-form.com/advanced-usage#ControlledmixedwithUncontrolledComponents), the two ways to register a controlled component would be:
- _register_ + _setValue_
- _Controller_ or _useController_

I was able to implement the first alternative, but I cannot for the life of me implement _Controller_ or, preferably, _useController_ with controlled components. Here's a [dummy application](https://codesandbox.io/s/sad-franklin-qk9k0) with both tries. 

What am I missing?

**Comment by bluebill1049** [comment]
What's the issue you got? looks fine to me.

**Reply by apedretti** [reply]
Hey there Bill, thank you for the package btw.

The issue I'm facing is that with both _Controller_ and _useController_ I can register a controlled component (using a hidden input) but I cannot control its value on RHF. [See here](https://i.imgur.com/8yAwucl.png).

**Comment by leapful** [comment, marked as answer]
@apedretti 

The ControlledComponentOne and ControlledComponentTwo in your sample are not handling on change to update field value in react-hook-form. The "field" prop of useController or Controller are injected to an hidden input while the select action of those 2 components is from "onClick" of each flavor item (div element). That's why nothing happens and no value is set in react-hook-form.

Working sample: https://codesandbox.io/s/brave-borg-m7qr0?file=/src/ControlledComponentTwo.js

P.S: The working sample is not using hidden input because it's redundant for this use case with react-hook-form.

**Reply by apedretti** [reply]
Thank you so much! I've been struggling with this one for the last few days. Just implemented it in one of the custom components I have and it's working wonders. Thanks.

**Reply by bluebill1049** [reply]
Thank you very much @leapful ❤️
